state/allwatcher.go

```diff
@@ -272,6 +272,7 @@ func (m *backingMachine) updated(st *State, store *multiwatcherStore, id string)
 		info.InstanceStatus = oldInfo.InstanceStatus
 		info.InstanceId = oldInfo.InstanceId
 		info.HardwareCharacteristics = oldInfo.HardwareCharacteristics
+		info.CharmProfiles = oldInfo.CharmProfiles
```

Comment:
Would we rather have no data or old data here?  No data might be better?  Then the worker can handle it.

Reply:
So no data could be a no-op. OK, that makes sense